File: src/main/java/org/arif/DAY_04/longestCommonPrefix.js
const longestCommonPrefix = function(strs) {
    if (strs.length === 0) return "";

    let answer = strs[0];
    for (let i = 1; i < strs.length; i++) {
        console.log(strs[i].indexOf(answer));
        while (strs[i].indexOf(answer) !== 0) {
            answer = answer.substring(0, answer.length - 1);
            if (answer === "") {
                return "";
            }
        }
    }
    return answer;
};

const strs = ["flower", "flower", "good"];
const answer = longestCommonPrefix(strs);
console.log(answer);

File: src/main/java/org/arif/DAY_04/LongestCommonPrefix.java
package org.arif.DAY_04;

import java.util.Arrays;

public class LongestCommonPrefix {
    public static void main(String[] args) {

        String[] strs = {"flower", "flow", "flight"};
        String result = longestCommonPrefix(strs);
        System.out.println(result);
    }

    public static String longestCommonPrefix(String[] strs) {
        for (int i = 0; i < strs[0].length(); i++) {
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].length() <= i || strs[j].charAt(i) != strs[0].charAt(i)) {
                    return strs[0].substring(0, i);
                }

            }
        }
        return strs[0];
    }

    public static String longestCommonPrefix02(String[] strs) {
        if (strs == null || strs.length == 0) return "";
        Arrays.sort(strs);
        String first = strs[0];
        String last = strs[strs.length - 1];

        int index = 0;
        while (index < first.length()) {
            if (first.charAt(index) == last.charAt(index)) {
                index++;
            } else {
                break;
            }

        }

        return first.substring(0, index);
    }
}
